e2e/scripts/st_file_uploader.py

```diff
@@ -16,4 +16,17 @@
 import streamlit as st
 
 
-st.file_uploader("Drop a file:", type=["txt"])
+result = st.file_uploader("Drop a file:", type=["txt"], accept_multiple_files=False)
+if result is not None:
+    st.text(result.getvalue())
+else:
+    st.text("No upload")
+
+result = st.file_uploader(
+    "Drop multiple files:", type=["txt"], accept_multiple_files=True
+)
+if result is not None:
+    strings = sorted([s.getvalue() for s in result])
+    st.text("\n".join(strings))
+else:
+    st.text("No upload")
```

Comment:
I prefer leaving an end-to-end test here (this doesn't rely on screenshot comparisons, so it shouldn't be flaky), because it tests the literal end-to-end flow of uploading a file (frontend), receiving the file (server), and re-displaying it (frontend).

Reply:
+1